docs/learn/multi-format-reader.md

```diff
@@ -0,0 +1,263 @@
+    "@data": data_callback if data_callback is not None else self.identity,
+    "@eln": eln_callback if eln_callback is not None else self.identity,
+}
+```
+That means, if the config file has an entry ```{"/ENTRY/title": "@attrs:metadata/title"}```, the `get_attr` method of the reader gets called and should return an attribute from the given path, i.e., in this case from `metadata/title`.
+
+By default, the MultiFormatReader supports the following special prefixes:
+
+- `@attrs`: To get metadata from the read-in experiment file(s). You need to implement the `get_attr` method in the reader.
+- `@data`: To get measurement data from the read-in experiment file(s). You need to implement the `get_data` method in the reader.
+- `@eln`: To get metadata from addtional ELN files. You need to implement the `get_eln_data` method in the reader.
+- `@link`: To implement a link between two entities in the NeXus file. By default, the link callback returns a dict of the form {"link": value.replace("/entry/", f"/{self.entry_name}/")}, i.e., a generic `/entry/` get replaced by the actual `entry_name`.
+
+In addition, the reader can also implement the `get_data_dims` method, which is used to return a list of the data dimensions (see below for more details).
+
+All of `get_attr`, `get_data`, and `get_eln_data`  (as well as any similar method that might be implemented) should have the same call signature:
+```python
+def get_data(self, key: str, path: str) -> Any:
+```
+Here, `key` is the config dict key (e.g., `"/ENTRY[my-entry]/data/data"`) and path is the path that comes _after_ the prefix in the config file. In this example, `path` would be `mydata`. With these two inputs, the reader should be able to return the correct data for this template key.
+
+### Special rules
+- **Lists as config value**: It is possible to write a list of possible configurations of the sort
+  ```json
+  "/ENTRY/title":"['@attrs:my_title', '@eln', 'no title']"
+  ```
+  The value most be a string of a list, with each item being a string itself. This allows to provide different options depending if the data exists for a given callback. For each list item , it is checked if a value can be returned and if so, the value is written. In this example, the converter would check (in order) the `@attrs` (with path `"my_title"`) and `@eln` (with path `""`) tokens and write the respective value if it exists. If not, it defaults to "no title".
+  This concept can be particularly useful if the same config file is used for multiple measurement configurations, where for some setup, the same metadata may or not be available.
+
+    Note that if this notation is used, it may be helpful to pass the `suppress_warning` keyword as `True` to the read function. Otherwise, there will be a warning for every non-existent value.
+
+- **Wildcard notation**: There exists a wildcard notation (using `*`)
+  ```json
+  "/ENTRY/data/AXISNAME[*]": "@data:*.data",

```

Comment:
I remember a discussion, that UPPERCASE notation is actually not valid for field names. What is the current status here? While the converter does not complain, it might just ignore it (?)

Reply:
This can only for the `NXdata` where we have `AXISNAME` and `DATA` and there is some more logic (like `signal`, `axes` and so on) to distinguish the two. The verification handles this special case differently. Other than that, it is indeed not allowed.